serializer.go

```diff
@@ -133,24 +142,29 @@ func processYamlFile(fileInfo os.FileInfo, yamlDir string, registerName string)
 func processYaml(yamlFile io.Reader, registerName string) {
 	var contentJson string
 	var err error
+	var key string
 
 	switch registerName {
 	case "datatype":
-		var dt Datatype
-		yaml.Unmarshal(streamToBytes(yamlFile), &dt)
-		contentJson, err = toJsonStr(dt)
+		var r Datatype
+		yaml.Unmarshal(streamToBytes(yamlFile), &r)
+		contentJson, err = toJsonStr(r)
+		key = getKeyFromStruct(reflect.ValueOf(&r), registerName)
```

Comment:
So I could just hard-code the names of the registers in each of the `case` blocks to the specific register in question, however at the time my goal was to completely get rid of this `switch` and get the actual datatype (i.e. `Register`, `Datatype` etc) dynamically.

If you would rather pull out each of the key values directly out of the datatype then I'm happy to do this.